### PR TITLE
shows a banner alerting the user if they are using Safari

### LINKED ISSUE #94: Have Marvin-web issue a warning or banner if the browser is not supported
If we are not going to support Safari (ains), we should at least show some kind of warning when entering the web with that browser, so that the users know to use a different one.

### PR DESCRIPTION
**Pull request by albireox**
This PR fixes #94, by showing a closable banner when the user is using Safari. The banner looks like

screen shot 2016-12-10 at 10 43 55

with the *Learn More* link taking you to the [known issue](https://api.sdss.org/doc/manga/marvin/known_issues.html#known-browser) about Safari. Right now it will show up in all versions of Safari. We could remove it for users using Safari Technology Preview, but since that version is still beta and we are not sure is fully compatible with Marvin, I would leave it. The banner is set to not reappear for a day. This may be a bit annoying. Maybe three or five days is better?

### CODE REVIEW

File: python/marvin/web/templates/header.html [hunk 1]
```diff
@@ -1,6 +1,35 @@
 
 {% from 'macros.html' import galidsearch, login %}
 
+
+<!-- Begin Cookie Consent plugin by Silktide - http://silktide.com/cookieconsent.
+We use it to show a banner if the used browser does not play well with Marvin.-->
+<link rel="stylesheet" type="text/css" href="//cdnjs.cloudflare.com/ajax/libs/cookieconsent2/3.0.3/cookieconsent.min.css" />
+<script src="//cdnjs.cloudflare.com/ajax/libs/cookieconsent2/3.0.3/cookieconsent.min.js"></script>
+<script>
+window.addEventListener("load", function(){
```
Comment (havok2063) [inline python/marvin/web/templates/header.html]: I would move this code marvin.js

**Comment by havok2063** [comment]
I see you've finally gotten your hands dirty with the web.  Good!  It looks like this cookieconsent code is primarily used to inform people cookies, which we're using as generic banner code.  That should be ok. At the risk of causing you to swear off the web and curse my name, I would suggest moving this code into the marvin.js file and generalizing it so we could potentially create other banners.  

Also, it looks like detecting browsers and features within browsers has gotten better, and people recommend using Modernizer instead. https://modernizr.com/ If we need to do more browser things,  we might want to switch to this.

File: python/marvin/web/templates/header.html [hunk 2]
```diff
@@ -1,6 +1,35 @@
...
+    if (navigator.userAgent.indexOf('Safari') != -1 && navigator.userAgent.indexOf('Chrome') == -1) {
```
Comment (havok2063) [inline python/marvin/web/templates/header.html]: alternative to navigator.userAgent, jquery has a browser attribute, accessible via $.browser.   $.browser=='safari' I think should work.  Might clean up the code.

File: python/marvin/web/templates/header.html [hunk 3]
```diff
@@ -1,6 +1,35 @@
...
+        window.cookieconsent.initialise({
```
Comment (havok2063) [inline python/marvin/web/templates/header.html]: can we turn this cookieconsent intialization into a general banner creation, and pass in as variables, the message, dismiss, and href options content?

File: python/marvin/web/templates/header.html [hunk 4]
```diff
@@ -1,6 +1,35 @@
...
+            "popup": {
+              "background": "#000"
+            },
+            "button": {
+              "background": "#f1d600"
+            }
+          },
+          "position": "top",
+          "cookie": {
+              "expiryDays": 1},
```
Comment (havok2063) [inline python/marvin/web/templates/header.html]: Does this mean a cookie is created which expires after 1 day?  Does this mean the banner only displays at most once a day?  Is this the behavior we want?  That seems reasonable to me, but just want to double-check.

**Comment by albireox** [comment]
Well, there has been a fair amount of swearing, that's for sure. I've moved the code to loayout and to utils.js under a function `marvinBanner`. The Safari banner is now called in `index.html`. I haven't been able to use your `$.browser=='safari'` so I went back to use the same I had.

In `marvinBanner` I have added a few configuration options, but that said, I'm not sure how well this will work as a general banner solution. By default, the library always creates a cookie, so the only way of making the banner work every time you open a certain page is to remove the cookie after it has been created. There are potential issues with domain and path in there. So, if at some point we end up needing a banner solution, there are probably other libraries out there. But for the purposes of this fix, this one works.

For reference, the API documentation of the library is in http://cookieconsent.wpengine.com/documentation/javascript-api/.

**Comment by havok2063** [comment]
The $.browser may be deprecated in jquery, not sure.   Ok, I think this looks reasonable enough.  I've had a look through the docs and yeah, it doesn't seem like this is a general all-purpose banner.  But it should suffice for now.

**Pull request merged**